js_modules/dagster-ui/packages/ui-core/src/workspace/WorkspaceContext/util.ts

```diff
@@ -218,3 +221,54 @@ export function repoLocationToRepos(repositoryLocation: RepositoryLocation) {
     return {repository, repositoryLocation};
   });
 }
+
+export function mergeWorkspaceData(
+  workspaceLocationEntry: PartialWorkspaceLocationNodeFragment,
+  workspaceLocationAssetsEntry: WorkspaceLocationAssetsEntryFragment,
+) {
+  const result: Partial<WorkspaceLocationNodeFragment> = {};
+  const {locationOrLoadError, ...rest} = workspaceLocationEntry;
+  Object.assign(result, rest);
```

Comment:
Could maybe just do this, to skip the `Object.assign`?

```
const {locationOrLoadError, ...result} = workspaceLocationEntry;
```

and go from there, maybe without even having to cast with the `Partial`